### ISSUE TITLE
[Scaler] General procedure. [SHAP] Single timestamp forceplots. 

### ISSUE THREAD

**Issue by nomadicsenseis** [unit8co/darts#1985]
Hi there! First of all, congratulations on the amazing library; I'm really enjoying it.

I've been using Darts for my latest project (so by no means am I an expert), and I've come across a couple of questions that I would greatly appreciate your help with.

I've used the standard TimeSeries.from_series() method to create three different TimeSeries objects from the columns of my original DataFrame. In the first one, I selected a single column as the target. However, I've also used this method to create TimeSeries objects for past_covariates and future_covariates, both of which correspond to multiple columns in my DataFrame. Is this a valid approach, or should I use the from_dataframe() method?

Regarding scaling, due to the difference in dimensions of my three TimeSeries objects, I've created three different Scaler() objects, each of which is fitted on the training time series object for the target, past_covariates, and future_covariates. So, each of them has a different scaling. To be honest, I'm not perfectly sure if this is the right approach. Are there any better practices?

About the historical_forecast() method, I've set the retrain=False parameter to check if my model is overfitting. I must admit I haven't been able to figure out the proper math behind selecting the stride and start parameters based on the output_chunk_length and lags of my model. But more importantly, if I leave retrain=True, I was not able to scale the corresponding training set accordingly to avoid data leakage.

It would be really nice if the gridsearch() method allowed for experimenting with different time-related hyperparameters (meaning lags) for each of the RegressionModel() regressors. That way, one could experiment in a grid-search manner with different combinations of lags, models, and hyperparameters for those models.

About the explainability AI module, as I mentioned, I scaled my data before fitting my model. With the backtest and predictions, I could inverse_transform() my outputs, but I was wondering if I could do the same thing with my explanations. In particular, I'd love to have the force_plot_from_ts() over my original scaled data.

Is there any way to choose force_plots() for single timestamps?

I know this is a lot, so thank you for reading all the way. I hope not to be too much of a bother. If there's any other channel where I can post these questions, I'd be happy to reach out there as well. :)

**Comment by madtoinou** [comment]
Hi @nomadicsenseis,

I am going to answer your questions point by point:
- `from_dataframe` will create one `TimeSeries`, it would then require some manipulation to isolate the `target` from the `past/future_covariates`. Your approach seems reasonable.
- fitting one scaler for each series is an acceptable approach, but it will ultimately depends on the nature of your dataset.
- #1540 tracks the feature you're describing : adding an argument to `historical_forecasts()` so that model can be retrained while applying a data-transformer on the series without leaking information.
- you can already include the `lags` in the `parameters` argument of `gridsearch()`
- What kind of information do you expect to get from the force plot with the un-normalized data? You could run the explainability on the unscaled data but I expect the model to struggle with it...
- As mentioned in your other issue, you need to slice the foreground series before passing it to the explainability model in order to restrict the number of timestamps, I don't think any other way but I will verify.
- We also answer questions on the gitter channel [here](https://app.gitter.im/#/room/#u8darts_darts:gitter.im?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) but I think that it's better to open issues when the questions are well structured and "deep" (github is easier to naviguate and query than gitter for other users)
